include/dialog_control.h

```diff
@@ -0,0 +1,72 @@
+#ifndef DIALOG_CONTROL_H
+#define DIALOG_CONTROL_H
+
+#include "dialog_ids.h"
+#include "course_table.h"
+
+// General string macros
+#define STRINGIFY(s) #s
+#define TO_STRING(s) STRINGIFY(s)
+#define HEX(s)       GLUE2(0x, s)
+#define HEX_STR(s)   TO_STRING(GLUE2(\x, s))
```

Comment:
rename the string macros to show that they aren't doing different things, and move these in the file with GLUE2